### ISSUE TITLE
'No problems found' on the files been detected as Virus by ClamAV

### ISSUE THREAD

**Issue by PeterDaveHello** [Maikuolan/phpMussel#215]
### phpMussel version used

The current latest v2 branch aab251349999722b450bc10f30c524bb7c1eafe3 (Two commits ahead of v2.1.2 release tag)

### Signatures version

I tested with two sources of signatures, both the latest of them:

1. manually installed:

- daily.cld version: 25775, sigs: 2249665
- main.cld version: 59, sigs: 4564902
- bytecode.cld version: 331, sigs: 94

2. fetched from [phpMussel/Signatures](https://github.com/phpMussel/Signatures)

- https://github.com/phpMussel/Signatures/commit/5c8bb8f1dccb7b50b78fb53ba151f623d648ca94 (2020.66.707/2020.66.708/2020.72.195)

### PHP SAPI used

1. cli
2. cgi-fcgi

### PHP version installed

v7.2.25

### Operating system used

Ubuntu 16.04.6 xenial LTS
(Linux sun 4.4.0-169-generic #198-Ubuntu SMP Tue Nov 12 10:38:00 UTC 2019 x86_64)

---

## Issue

I got some virus samples that would be detected as virus by ClamAV, but can't be detect by phpMussel, they're actually html files contains VBscript, using `.hta` as filename extension

### Expected behavior

The file should be detected as virus, as it's marked as virus by ClamAV, and more than 10+ virus scan engines on VirusTotal

### Actual behavior

phpMussel says `No problems found.` about the file

### Steps to reproduce the behavior

* Simply scan the file using phpMussel cli or frontend

### Last known working version without this issue.

* N/A

---

### Relevant scan_log entry

```
Wed, 08 Apr 2020 19:18:51 +0800 Started.
> Checking 'test.hta' (FN: 1218b3be; FD: 70be64db):
-> No problems found.
Wed, 08 Apr 2020 19:18:52 +0800 Finished.
```

### Scan result from ClamAV Scanner 0.102.2

```
Scanning test.hta
test.hta: Win.Trojan.VBAgent-6647891-0 FOUND

----------- SCAN SUMMARY -----------
Known viruses: 6804144
Engine version: 0.102.2
Scanned directories: 0
Scanned files: 1
Infected files: 1
Data scanned: 0.02 MB
Data read: 0.02 MB (ratio 1.00:1)
Time: 13.659 sec (0 m 13 s)
```

### VirusTotal Scan result

https://www.virustotal.com/gui/file/91a1fe03d7ed5444927de1c87bd0a4f625720edf854f0d3c1f985cb602263dfc/detection

(detected by 16 of 58 engines)

Let me know that if you need the files to dig deeper, thanks a lot!

**Comment by Maikuolan** [comment]
Hi Peter,

Sorry for the delayed reply, and thanks for your patience. :-)

I haven't looked into this deeply yet, but I'll look into it within the next few hours and reply back here to let you know what I discover.

**Comment by Maikuolan** [comment]
So, I've found the reason why phpMussel isn't detecting that particular sample, despite ClamAV detecting it.

Per the report from Virus Total, the name of the signature responsible for enabling ClamAV to detect that particular sample is "**Win.Trojan.VBAgent-6647891-0**".

Manually searching the ClamAV database, I find that particular signature located in the `daily.ldb` signature file:
```
Win.Trojan.VBAgent-6647891-0;Engine:55-255,Target:3;0&1;7265706c616365{-500}7265706c616365;636872286362797465{-100}6d696428{-500}636872286362797465{-100}6d696428
```

Per [the current documentation for phpMussel's SigTool](https://github.com/phpMussel/SigTool/blob/master/_docs/readme.en.md#note-regarding-signature-file-extensions):
> - __ldb__: Logical signature files (these will *eventually*, for a future SigTool version, be extracted from the `.ldb` signature files contained by `daily.cvd` and `main.cvd`, but aren't yet supported by SigTool or phpMussel). Signature files that work with various logical rules use this extension.

In short, the signature responsible is one of ClamAV's "logical" signatures (the **ldb** signatures), and I don't yet have a reliable way of translating ClamAV's ldb signatures into something that phpMussel can handle properly. Therefore, there isn't currently any reliable equivalent or analogue to that particular ClamAV signature available for phpMussel.

I do hope for phpMussel to *eventually* be able to support ClamAV's ldb signatures, but I don't have any particular deadline or schedule for it at this time.

Possibly, to be able to catch those samples in the meantime, I could write some equivalents/analogues to that particular signature for phpMussel manually and add them to one of phpMussel's signature files. I'll investigate that possibility now.

**Comment by Maikuolan** [comment]
> Possibly, to be able to catch those samples in the meantime, I could write some equivalents/analogues to that particular signature for phpMussel manually and add them to one of phpMussel's signature files. I'll investigate that possibility now.

..and done. :-)

(( https://github.com/phpMussel/Signatures/commit/f413bf6ab81bbcd78734b693cefe50e0bac401dd ))

```
$is_html:1;FD-NORM-RX:7265706c616365.{0,1000}7265706c616365;FD-NORM-RX:636872286362797465.{0,200}6d696428.{0,1000}636872286362797465.{0,200}6d696428; VBAgent-6647891-0.Equiv
```

**Comment by Maikuolan** [comment]
Updating to that latest update should resolve the problem. Let me know whether the problem persists. Cheers. :-)

**Comment by PeterDaveHello** [comment]
Hi @Maikuolan,

Thanks a lot for the prompt reply and explanation, is the change committed to the SigTool or phpMussel itself so that the future updates will also be applied? It's a little bit interesting that I updated the signatures locally, now the cli can detect the virus, but the web front-end can, I'm not sure if I can manually use the SigTool to generate the latest signatures as there seems to be no new commits there yet.

Thank you again!

**Comment by Maikuolan** [comment]
> is the change committed to the SigTool or phpMussel itself so that the future updates will also be applied?

This particular change was just committed to the `clamav.cedb` signature file (that particular signature file is a little different to the other signature files derived from the ClamAV database, in the sense that unlike the others, I update that one manually in a text editor, rather than by using SigTool or anything else). So, no changes done to SigTool or phpMussel itself yet. :-)

I still plan to eventually update both SigTool (to be able to properly process the **ldb** signatures into something that phpMussel can handle properly) and phpMussel itself (accordingly, to be able to handle such signatures) at some point in the future, but I need to plan it out properly first.

> now the cli can detect the virus, but the web front-end

It could be a caching issue, maybe? It might work the same for both after clearing phpMussel's cache, maybe (but, I'm only speculating, and it's possible that this difference might be caused by something else; a caching issue is the first thing that comes to my mind though).

> I'm not sure if I can manually use the SigTool to generate the latest signatures as there seems to be no new commits there yet.

It should be possible to use SigTool to generate your own signature files, as a means to keep phpMussel's signatures up-to-date with the latest ClamAV signatures database.

I recorded a [video demonstration for SigTool](https://www.youtube.com/watch?v=f2LfjY1HzRI), which could help (I recorded it a few years ago, so it's a little old now, but should mostly still be relevant).

But yeah; I'll still need to sort out a way for SigTool to be able to properly process **ldb** signatures, seeing as it can't do that yet.

**Comment by PeterDaveHello** [comment]
> This particular change was just committed to the `clamav.cedb` signature file (that particular signature file is a little different to the other signature files derived from the ClamAV database, in the sense that unlike the others, I update that one manually in a text editor, rather than by using SigTool or anything else). So, no changes done to SigTool or phpMussel itself yet. :-)

Can I consider `clamav.cedb` a manually phpMussel maintained signature file?

> I still plan to eventually update both SigTool (to be able to properly process the **ldb** signatures into something that phpMussel can handle properly) and phpMussel itself (accordingly, to be able to handle such signatures) at some point in the future, but I need to plan it out properly first.

Sounds very great! Let me know when it's ready to be tested and I'll be happy to help with my spare time.

> It could be a caching issue, maybe? It might work the same for both after clearing phpMussel's cache, maybe (but, I'm only speculating, and it's possible that this difference might be caused by something else; a caching issue is the first thing that comes to my mind though).

Looks like cache issue maybe, though I can see the signature version is up-to-date on the web front-end, anyway, I just tested it again, and it's working now, awesome!

> It should be possible to use SigTool to generate your own signature files, as a means to keep phpMussel's signatures up-to-date with the latest ClamAV signatures database.
> 
> I recorded a [video demonstration for SigTool](https://www.youtube.com/watch?v=f2LfjY1HzRI), which could help (I recorded it a few years ago, so it's a little old now, but should mostly still be relevant).

No problem, it's pretty handy to use it without problem, just asked because I'm not sure if this update made by users will keep the changes from https://github.com/phpMussel/Signatures/commit/f413bf6ab81bbcd78734b693cefe50e0bac401dd or not, means up-to-date with ClamAV DB but also keep the manual added signatures.
